src/Collection.js

```diff
@@ -51,14 +63,16 @@ var Collection = React.createClass({
 			var contents = <table className="table table-striped table-hover table-condensed">
 				<thead><tr>
 				{Object.keys(this.state.schema.properties).map(function(name, i) {
-					return <td key={name}>{name}</td>
+					return <td key={name}>{name.replace(/_/g, ' ')}</td>
 				})}
 				</tr></thead>
 				<tbody>
 					{data.map(function(row, i) {
 						return <tr key={i}>
 							{Object.keys(this.state.schema.properties).map(function(name, i) {
-								var text = typeof row[name] == "undefined" || row[name].length < 50 ? row[name] : row[name].substring(0, 45)+'...';
+								var text = row[name] != undefined ? String(row[name]) : ''
+								if( text.length && text[0].match(/^[\[{]/) != null ) text = jsonToHuman(text)
```

Comment:
Agreed, that looks better.
I guess my idea was to not alter the code too much, but you definately have a point that it looks complex.